### PR TITLE
Update go-zfs to last version

### PR DESCRIPTION
**Pull request by kvasdopil**
FreeBSD build requires last updates to go-zfs to be included

### CODE REVIEW

File: hack/vendor.sh [hunk 1]
```diff
@@ -49,7 +49,7 @@ clone git github.com/go-check/check 64131543e7896d5bcc6bd5a76287eb75ea96c673
 clone git github.com/gorilla/context 14f550f51a
 clone git github.com/gorilla/mux e444e69cbd
 clone git github.com/kr/pty 5cf931ef8f
-clone git github.com/mistifyio/go-zfs v2.1.0
+clone git github.com/mistifyio/go-zfs cdc0f941c4d0e0e94d85348285568d921891e138
```
Comment (tianon) [inline hack/vendor.sh]: Can we instead ask upstream to tag a new version?

Comment (kvasdopil) [inline hack/vendor.sh]: Asked them about that.

Comment (kvasdopil) [inline hack/vendor.sh]: Done

**Comment by calavera** [comment]
LGTM

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**